packages/ketcher-core/src/application/render/restruct/restruct.ts

```diff
@@ -309,6 +309,33 @@ class ReStruct {
     })
   }
 
+  getSelectionRotationCenter(
+    selection: Partial<Record<keyof typeof ReStruct.maps, number[] | undefined>>
+  ): Vec2 {
+    let vbox: Box2Abs | null = null
+    // we can't use just getVBoxObj from atom!
+    // because this will lead incorrect center position for the move atom around it
+    // because of atom's vBox contain text label with is not constant after flip/rotate
+    for (const atomId of selection.atoms ?? []) {
+      const atomPP = this.atoms.get(atomId)!.a.pp
+      const atomBox = new Box2Abs(atomPP, atomPP)
+      vbox = vbox == null ? atomBox : Box2Abs.union(vbox, atomBox)
```

Comment:
If you understood what vbox is may be it is worth adding more sensible name?
How does it compare with bounding box?

Reply:
fixed